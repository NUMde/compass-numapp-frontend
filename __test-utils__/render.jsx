/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

import UserReducer from '../src/store/user.slice';
import GlobalsReducer from '../src/store/globals.slice';
import QuestionnaireReducer from '../src/store/questionnaire.slice';

const testReducer = combineReducers({
  User: UserReducer,
  Globals: GlobalsReducer,
  Questionnaire: QuestionnaireReducer,
});

function renderWithRedux(
  ui,
  {
    initialState = {
      User: UserReducer.initialState,
      Globals: GlobalsReducer.initialState,
      Questionnaire: QuestionnaireReducer.initialState,
    },
    store = createStore(testReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

export { renderWithRedux };
