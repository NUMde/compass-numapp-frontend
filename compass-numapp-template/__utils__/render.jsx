import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react-native';
import { buildStore } from './mockStore';

let store = null;

export function renderWithRedux(subComponent, state = {}) {
  if (!store) {
    store = buildStore(state);
  }

  const queries = render(<Provider store={store}>{subComponent}</Provider>);

  return {
    ...queries,
    store,
  };
}

export function initRenderModule(initialState = {}) {
  store = buildStore(initialState);

  return store;
}
