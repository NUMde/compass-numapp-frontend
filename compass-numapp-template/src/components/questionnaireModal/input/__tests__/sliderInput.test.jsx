import React from 'react';

import { renderWithRedux, act } from '__test-utils__/render';

import SliderInput from '../sliderInput';

const initialState = {
  Questionnaire: {
    itemMap: {
      1: {
        linkId: '1',
        type: 'group',
        text: 'group',
        item: [{ linkId: '1.1', type: 'integer' }],
      },
      1.1: { linkId: '1.1', type: 'integer', text: 'integer' },
    },
  },
};

const item = {
  linkId: '1.1',
  type: 'integer',
  text: 'integer',
  extension: [
    {
      url: 'http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl',
      valueCodeableConcept: {
        coding: [
          {
            system: 'http://hl7.org/fhir/questionnaire-item-control',
            code: 'slider',
          },
        ],
      },
    },
    {
      url: 'https://num-compass.science/fhir/StructureDefinition/LowRangeLabel',
      valueString: 'Low-Label',
    },
    {
      url: 'https://num-compass.science/fhir/StructureDefinition/HighRangeLabel',
      valueString: 'High-Label',
    },
    {
      url: 'http://hl7.org/fhir/StructureDefinition/questionnaire-sliderStepValue',
      valueInteger: 1,
    },
    {
      url: 'http://hl7.org/fhir/StructureDefinition/minValue',
      valueInteger: 0,
    },
    {
      url: 'http://hl7.org/fhir/StructureDefinition/maxValue',
      valueInteger: 10,
    },
  ],
};

describe('SliderInput', () => {
  it('should render sliderInput with all properties', () => {
    const { getByText, toJSON } = renderWithRedux(<SliderInput item={item} />, {
      initialState,
    });
    expect(toJSON()).toMatchSnapshot();
    expect(getByText('Low-Label')).toBeTruthy();
    expect(getByText('High-Label')).toBeTruthy();
  });

  it('should update value when sliding complete', () => {
    const { getByTestId } = renderWithRedux(<SliderInput item={item} />, {
      initialState,
    });
    expect(getByTestId('Slider')).toBeTruthy();
    act(() => {
      getByTestId('Slider').props.onRNCSliderSlidingComplete({
        nativeEvent: { value: 7 },
      });
    });
    expect(getByTestId('Slider').props.value).toBe(7);
  });
});
