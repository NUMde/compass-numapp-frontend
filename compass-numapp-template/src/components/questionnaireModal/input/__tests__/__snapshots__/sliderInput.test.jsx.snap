// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderInput should render sliderInput with all properties 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 33.75,
        "lineHeight": 43.125,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
  >
    integer
  </Text>
  <RNCSlider
    accessibilityHint="0 equals Low-Label, and 10 equals High-Label"
    disabled={false}
    enabled={true}
    inverted={false}
    maximumTrackTintColor="#1F2532"
    maximumValue={10}
    minimumTrackTintColor="#1F2532"
    minimumValue={0}
    onChange={null}
    onRNCSliderAccessibilityAction={null}
    onRNCSliderSlidingComplete={[Function]}
    onRNCSliderSlidingStart={null}
    onRNCSliderValueChange={null}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    step={1}
    style={
      [
        {
          "width": 710,
        },
        {
          "height": 40,
        },
      ]
    }
    tapToSeek={false}
    testID="Slider"
    value={5}
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 710,
      }
    }
  >
    <Text
      style={
        {
          "textAlign": "left",
          "width": "33%",
        }
      }
    >
      Low-Label
    </Text>
    <Text
      style={
        {
          "textAlign": "right",
          "width": "33%",
        }
      }
    >
      High-Label
    </Text>
  </View>
</View>
`;
